components/logins/src/sync/payload.rs

```diff
@@ -61,44 +113,19 @@ pub struct LoginPayload {
 
     #[serde(default)]
     pub times_used: i64,
+
+    // Additional "unknown" round-tripped fields
+    #[serde(flatten, skip_serializing_if = "serde_json::Map::is_empty")]
```

Comment:
This is quite magic 😮

Reply:
This is definitely cool.  Does `flatten` apply to serializing as well as deserializing?  If so, is the `skip_serializing_if` needed?

Reply:
Good question :) I expect the difference might be the key missing vs existing as `{}`; the former seems preferable even if the same in practice? Needs tests to clarify though.